server/webapp/WEB-INF/rails.new/app/assets/new_javascripts/models/agents/environments.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2016 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+define(['mithril', 'lodash', 'helpers/mrequest', 'models/agents/tri_state_checkbox'], function (m, _, mrequest, EnvironmentCheckbox) {
+  var Environments  = {};
+  Environments.list = [];
+
+  var isPresentOnAll = function (selectedAgents, value) {
+    return _.every(selectedAgents, function (agent) {
+      return _.includes(agent.environments(), value);
+    })
+  };
+
+  var isPresentOnAny = function (selectedAgents, value) {
+    return _.some(selectedAgents, function (agent) {
+      return _.includes(agent.environments(), value);
+    })
+  };
```

Comment:
If you extract `.environments()` and `.resources()` (from the file below) you may be able to extract `isPresentOn..` methods into a common codebase (probably inside `tri_state_checkbox.js`)